src/main/java/appeng/client/render/cablebus/P2PTunnelFrequencyModel.java

```diff
@@ -0,0 +1,43 @@
+
+package appeng.client.render.cablebus;
+
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.function.Function;
+
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.model.IModel;
+import net.minecraftforge.common.model.IModelState;
+
+import appeng.core.AppEng;
+
+
+public class P2PTunnelFrequencyModel implements IModel
+{
+	private static final ResourceLocation TEXTURE = new ResourceLocation( AppEng.MOD_ID, "parts/p2p_tunnel_sides" );
+
+	@Override
+	public Collection<ResourceLocation> getTextures()
+	{
+		return Collections.singletonList( TEXTURE );
+	}
+
+	@Override
+	public IBakedModel bake( IModelState state, VertexFormat format, Function<ResourceLocation, TextureAtlasSprite> bakedTextureGetter )
+	{
+		try
+		{
+			final TextureAtlasSprite texture = bakedTextureGetter.apply( TEXTURE );
+			return new P2PTunnelFrequencyBakedModel( format, texture );
+		}
+		catch( Exception e )
+		{
+			throw new RuntimeException( e );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00112)